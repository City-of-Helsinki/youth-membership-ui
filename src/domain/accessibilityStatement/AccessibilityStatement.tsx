import React from 'react';
import { useTranslation } from 'react-i18next';

import getLanguageCode from '../../common/helpers/getLanguageCode';
import PageContent from '../../common/components/layout/PageContent';
import styles from './AccessibilityStatement.module.css';
import AccessibilityStatementFi from './AccessibilityStatementFi';
import AccessibilityStatementEn from './AccessibilityStatementEn';
import AccessibilityStatementSv from './AccessibilityStatementSv';

function AccessibilityStatement() {
  const { i18n } = useTranslation();
  const selectStatement = () => {
    const lang = getLanguageCode(i18n.languages[0]);

    switch (lang) {
      case 'fi':
        return <AccessibilityStatementFi />;
      case 'en':
        return <AccessibilityStatementEn />;
      case 'sv':
        return <AccessibilityStatementSv />;
      default:
        return <p>Invalid language.</p>;
    }
  };
  return (
    <>
      <div className={styles.background} />
      <PageContent>
        <div className={styles.innerWrapper}>{selectStatement()}</div>
      </PageContent>
    </>
  );
}

export default AccessibilityStatement;
