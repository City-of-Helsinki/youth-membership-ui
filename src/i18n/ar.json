{
  "appName": "بطاقة العضو أي ياسّاري (jässäri)",
  "approval": {
    "addGuardianText": "بإمكاننا التواصل مع أولياء الأمر الآخرين الذين تم الإبلاغ عنهم بخصوص الأمور المتعلقة بالطفل.",
    "addInfo": "معلومات إضافية",
    "additionalAddresses": "العناوين الأخرى",
    "address": "عنوان المنزل",
    "approveButton": "موافق على العضوية",
    "approvedLink": "لقد أصبح الرابط قديمًا",
    "approvedMessage": "شكراً لك لتأكيدك على العضوية لدى خدمات الشباب.",
    "approvedTitle": "تمت الموافقة على طلب العضوية!",
    "approverAcceptance": "الموافقات المطلوبة من ولي الأمر",
    "approverEmail": "البريد الإلكتروني",
    "approverFirstName": "الاسم الشخصي",
    "approverInfo": "بيانات ولي الأمر",
    "approverInfoText": "بإمكاننا التواصل مع ولي الأمر بخصوص الأمور المتعلقة بالطفل.",
    "approverLastName": "اسم العائلة:",
    "approveTerms": "لقد اطلعت <0> على السرد بشأن التسجيل لهذه الخدمة </0> <1>وعلى السرد بشأن التسجيل للملف التعريفي لهلسنكي</1> <2>وعلى المعاملات المتعلقة بحماية المعلومات للمدينة</2>. من الممكن التواصل معي من خلال المعلومات التي قدمتها.",
    "approveTermsText_link": "شروط الخدمة",
    "basicInfo": "البيانات الأساسية",
    "birthDate": "تاريخ الميلاد",
    "explanationCheckStatus": "بإمكان الشاب التحقق من وضع الطلب بتسجيل الدخول لياسّاري. اطلب منه إرسال رسالة التأكيد من جديد، إذا لم تتم الموافقة على عضويته بعد.",
    "explanationError": "أما أن تكون قد وافقت على الطلب الذين تقدم به الشاب ابنكم أو قد حدث خطأ لدى خدمتنا.",
    "formExplanationText_1": "قد أرسل من أجل موافقتك طلب العضوية لخدمات الشباب لمدينة هلسنكي. العضوية مجانية. بالموافقة على الطلب،",
    "formExplanationText_2": "من الممكن قضاء وقتًا في بيت الشباب والمشاركة في مجموعات ممارسة الهوايات والدورات التعليمية والمخيمات يقوم مرشدو الشباب الحاصلون على تعليم بتوجيه الفعالية في بيوت الشباب. إذا كانت هناك أخطاء في المعلومات، فاطلب من الشاب تغيير المعلومات وأن يُرسل لك رسالة تأكيد جديدة.",
    "name": "الاسم",
    "phone": "الهاتف",
    "photoUsageApproved": "الإذن بالتصوير",
    "photoUsageApprovedNo": "لا",
    "photoUsageApprovedText": "من الممكن التقاط صور للطفل ويجوز استخدامها أثناء التواصل والتسويق لمدينة هلسنكي في وسائل الإعلام المطبوعة والإلكترونية. لا يُدفع تعويض عن التصوير أو استخدام الصور. الحق في استخدام الصور يكون ساري المفعول دائمًا، ويمس كل التواصل لمدينة هلسنكي والتواصل بخصوص التسويق.",
    "photoUsageApprovedYes": "نعم",
    "profile": "الملف الشخصي",
    "schoolInfo": "المدرسة والصف",
    "title": "موافق على العضوية"
  },
  "authentication": {
    "deviceTimeError": {
      "message": "صدق أو لا تصدق، ولكن ليس بإمكانك تسجيل الدخول لأن ساعة جهازك أخطأت التوقيت بقدر يزيد عن 5 دقائق. اضبط الساعة وجرّب من جديد."
    },
    "errorMessage": "حدث خطأ. حاول من جديد.",
    "errorTitle": "أها!",
    "permRequestDenied": {
      "message": "يتوجب عليك الموافقة على الحقوق التي طلبناها في Tunnistamo لاستخدامك هذه الخدمة. الرجاء محاولة تسجيل الدخول من جديد."
    }
  },
  "confirmSendingProfile": {
    "buttonText": "أرسل طلب جديد إلى ولي الأمر",
    "helpText": "لقد أرسلت العضوية بشكل مقبول إلى العنوان",
    "linkToShowSentData": "أعرض بيانات الطلب",
    "sendAgainHelpText": "تم إرسال رسالة التأكيد لديك من جديد إلى العنوان",
    "title": "تم إرسال الطلب إلى ولي أمرك!"
  },
  "edit": {
    "pageTitle": "عدّل بياناتك"
  },
  "emailApproval": {
    "content": "مرحبًا {{ youth_profile.approver_first_name }},<br /><br />{{ youth_profile.profile.first_name }} قد أرسل بشكل مقبول طلب العضوية لخدمات الشباب لمدينة هلسنكي. العضوية مجانية. بالموافقة على الطلب، {{ youth_profile.profile.first_name }} يُصبح من الممكن المجيء إلى بيت الشباب والحصول على ميزات لطيفة. إذا كانت هناك أخطاء في المعلومات، فاطلب من الشاب تغيير المعلومات وأن يُرسل لك رسالة تأكيد جديدة. اذهب للتأكيد على خدمة ياسّاري باستخدام هذا الرابط: <br /><br /><a href=\"https://jassari.hel.fi/approve/{{ youth_profile.approval_token }}\">https://jassari.hel.fi/approve/{{ youth_profile.approval_token }}</a><br /><br /><i> تم إرسال هذه الرسالة تلقائيًا من المنظومة. لا تُجب على هذه الرسالة، لأنه لا يتم تداول الإجابات.</i>",
    "heading": "{{ youth_profile.profile.first_name }} قد أرسل بشكل مقبول طلب العضوية لخدمات الشباب لمدينة هلسنكي."
  },
  "emailApproved": {
    "content": "أهلاً وسهلاً بكم كعضو لدى خدمات الشباب لمدينة هلسنكي! {{ youth_profile.approver_first_name }} قد وافق على عضويتك. بطاقة العضوية الإلكترونية الخاصة بك موجودة هنا: <a href=\"https://jassari.hel.fi\">https://jassari.hel.fi</a><br /><br /> معلومات إضافية عن عضوية خدمات الشباب وميزاتها من الممكن قراءتها من الرابط <a href=\"http://jassari.munstadi.fi/\">http://jassari.munstadi.fi/</a>.<br /><br /><i> تم إرسال هذه الرسالة تلقائيًا من المنظومة. لا تُجب على هذه الرسالة، لأنه لا يتم تداول الإجابات.</i>",
    "heading": "{{ youth_profile.approver_first_name }} قد وافق على عضويتك لدى خدمات الشباب."
  },
  "emailRenew": {
    "content": "مرحبًا {{ youth_profile.profile.first_name }}! عضويتك مع خدمات الشباب سوف ينتهي مفعولها. جدد عضويتك بسهولة هنا https://jassari.hel.fi/ معلومات إضافة عن عضوية خدمات الشباب وميزاتها من الممكن قراءتها من الرابط http://jassari.munstadi.fi/ <br /><br /> معلومات إضافية عن عضوية خدمات الشباب وعن ميزاتها من الممكن قراءتها من الرابط http://jassari.munstadi.fi/<br /><br /> <i> تم إرسال هذه الرسالة تلقائيًا من المنظومة. لا تُجب على هذه الرسالة، لأنه لا يتم تداول الإجابات.</i>",
    "heading": "{{ youth_profile.profile.first_name }} عضويتك مع خدمات الشباب سوف ينتهي مفعولها."
  },
  "externalLink": {
    "description": "يفتح في نافذة جديدة"
  },
  "fields": {
    "city": "مدينة",
    "email": "عنوان البريد الإلكتروني",
    "firstName": "الاسم الشخصي",
    "lastName": "اسم العائلة",
    "phoneNumber": "رقم الهاتف",
    "photoUsageApproved": "إذن بالتصوير",
    "postalCode": "الرمز البريدي",
    "schoolClass": "الصف",
    "schoolName": "اسم المدرسة",
    "streetAddress": "عنوان المكان (الشارع)"
  },
  "footer": {
    "accessibility": "سرد بشأن التوفر",
    "copyright": "مدينة هلسنكي",
    "feedback": "قدّم ردّك",
    "feedbackLink": "https://www.hel.fi/kulttuurin-ja-vapaa-ajan-toimiala/fi/yhteystiedot-ja-palaute/Anna+palautetta/",
    "privacy": "سرد لبيانات التسجيل",
    "reserveRights": "جميع الحقوق محفوظة",
    "terms": "شروط استخدام الخدمة"
  },
  "helsinkiLogo": "شعار هلسنكي",
  "LANGUAGE_OPTIONS": {
    "ARABIC": "العربية",
    "ENGLISH": "الإنجليزية",
    "ESTONIAN": "الإستونية",
    "FINNISH": "الفنلندية",
    "FRENCH": "الفرنسية",
    "RUSSIAN": "الروسية",
    "SOMALI": "الصومالية",
    "SWEDISH": "السويدية"
  },
  "language": {
    "ar": "العربية",
    "en": "English",
    "et": "eesti keel",
    "fi": "Suomi",
    "fr": "français",
    "ru": "Русский язык",
    "so": "af Soomaali",
    "sv": "Svenska"
  },
  "loading": "جاري التحميل ...",
  "login": {
    "buttonText": "تقدم بطلب العضوية",
    "findNearestService": "اعثر على أقرب مكان للشباب من خارطة الخدمات",
    "helpText": "بإمكانك من خلال بطاقة العضوية ياسّاري المشاركة في فعالية خدمات الشباب. العضوية تضمن لك أفعال لطيفة لتقوم بها، فبإمكانك التجول في بيوت الشباب والمشاركة في مجموعات ممارسة الهوايات والدورات التعليمية والمخيمات، والتقدم بطلب مساعدة لمشروع أو حتى تنظيم حدث خاص بك! يقوم بتوجيه الفعالية في بيوت الشباب مرشدون للشباب حاصلون على تعليم.",
    "helpTextUnderAge": "للأسف التقدم بطلب بطاقة العضوية ياسّاري إلكترونيًا يتطلب أن يكون العمر على الأقل 13 سنة. بإمكانك تحميل طلب بطاقة العضوية ياسّاري من هنا. قم بتعبئة الطلب مع ولي أمرك وأعده إلى المكان المخصص للشباب، وبذلك ستحصل على بطاقة العضوية الخاصة بك.",
    "linkForMembersText": "هل أنت عضو؟ سجّل الدخول",
    "pageTitle": "بطاقة العضو لفعاليات الشباب",
    "registrationForm": "https://nuorten.helsinki/uploads/2021/08/eea833f3-jasenkorttikaavake_2021_suomi.pdf",
    "registrationFormText": "حمّل الطلب",
    "return": "ارجع إلى الخلف",
    "title": "أطلب بطاقة العضوية ياسّاري!"
  },
  "membershipDetails": {
    "additionalInformation": "معلومات إضافية",
    "edit": "عدّل البيانات",
    "photoUsageExplanation": "من الممكن التقاط صور لي ويجوز استخدامها أثناء التواصل والتسويق لمدينة هلسنكي في وسائل الإعلام المطبوعة والإلكترونية. لا يُدفع تعويض عن التصوير أو استخدام الصور. الحق في استخدام الصور يكون ساري المفعول دائمًا، ويمس كل التواصل لمدينة هلسنكي والتواصل بخصوص التسويق.",
    "profileInformation": "البيانات الأساسية",
    "requiredPermissionsFromParent": "الموافقات المطلوبة من الوالد/الوالدة",
    "returnToFront": "ارجع إلى صفحة الواجهة",
    "text": "بإمكانك من خلال بطاقة العضوية ياسّاري المشاركة في فعالية خدمات الشباب. العضوية تضمن لك أفعال لطيفة لتقوم بها، فبإمكانك التجول في بيوت الشباب والمشاركة في مجموعات ممارسة الهوايات والدورات التعليمية والمخيمات، والتقدم بطلب مساعدة لمشروع أو حتى تنظيم حدث خاص بك! يقوم بتوجيه الفعالية في بيوت الشباب مرشدون للشباب حاصلون على تعليم.",
    "title": "البيانات الشخصية"
  },
  "membershipInformation": {
    "pageTitle": "بيانات العضوية",
    "renew": "عضوية جديدة",
    "renewSuccessMessage": "تم تجديد العضوية بنجاح!",
    "renewSuccessMessageMinor": "تم إرسال طلب التجديد بنجاح",
    "renewSuccessTitle": "تم بنجاح",
    "renewUnderage": "أرسل طلب التجديد إلى ولي الأمر",
    "showProfileInformation": "اعرض بيانات الملف الشخصي",
    "status": {
      "description": {
        "expired": "أرسل العضوية من جديد إلى ولي أمرك للموافقة عليها.",
        "pending": "الطلب بخصوص تجديد العضوية قد تم إرساله إلى العنوان {{email}}",
        "renewable": "أرسل العضوية من جديد إلى ولي أمرك للموافقة عليها.",
        "renewing": "الطلب بخصوص تجديد العضوية قد تم إرساله إلى العنوان {{email}}"
      },
      "title": {
        "membershipExpired": "لقد انتهت فترة سريان مفعول العضوية.",
        "pending": "تجديد العضوية في انتظار موافقة ولي الأمر",
        "renewable": "تنتهي فترة سريان مفعول العضوية {{date}}.",
        "renewing": "تنتهي فترة سريان مفعول العضوية {{date}}. التجديد في انتظار موافقة ولي الأمر."
      }
    },
    "title": "رقم بطاقة العضو {{number}}",
    "validUntil": "ساري المفعول حتى {{date}}."
  },
  "meta": {
    "description": "بإمكانك من خلال بطاقة العضوية ياسّاري المشاركة في فعالية خدمات الشباب. العضوية تضمن لك أفعال لطيفة لتقوم بها، فبإمكانك التجول في بيوت الشباب والمشاركة في مجموعات ممارسة الهوايات والدورات التعليمية والمخيمات، والتقدم بطلب مساعدة لمشروع أو حتى تنظيم حدث خاص بك! يقوم بتوجيه الفعالية في بيوت الشباب مرشدون للشباب حاصلون على تعليم."
  },
  "nav": {
    "helsinki": "هلسنكي",
    "menu": "قائمة الخيارات",
    "menuButtonLabel": "قائمة خيارات الملف الشخصي",
    "profile": "الملف الشخصي - هلسنكي",
    "signin": "سجّل الدخول",
    "signout": "سجّل الخروج",
    "skipToContent": "انتقل إلى المحتوى الرئيسي للصفحة"
  },
  "notification": {
    "closeButtonText": "أغلق",
    "defaultErrorText": "حدث خطأ ما. حاول من جديد بعد قليل.",
    "defaultErrorTitle": "خطأ"
  },
  "oidc": {
    "authenticating": "يتم التعرف على الهوية"
  },
  "privacyPolicy": {
    "descriptionLink": "https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/tietoa-helsingista/tietosuoja/",
    "helsinkiProfileLink": "https://www.hel.fi/static/liitteet-2019/Kaupunginkanslia/Rekisteriselosteet/Keha/S%C3%A4hk%C3%B6isten%20asiointipalveluiden%20rekisteri.pdf"
  },
  "profile": {
    "address": "العنوان",
    "email": "البريد الإلكتروني",
    "loading": "نُحمّل الملف الشخصي ...",
    "name": "الاسم",
    "phone": "رقم الهاتف",
    "verifying": "نتحقق من الملف الشخصي ..."
  },
  "registration": {
    "addAddress": "أضف عنوان آخر",
    "addGuardian": "أضف بيانات الاتصال لولي أمر آخر",
    "addGuardianText": "بإمكاننا عند الحاجة التواصل مع أولياء الأمر بخصوص الأمور المتعلقة بعضويتك مع خدمات الشباب.",
    "addInfo": "معلومات إضافية",
    "address": "عنوان المكان (الشارع)",
    "ageRestriction": "بطاقة العضو أي ياسّاري مُخصصة للشباب والشابات الذين تبلغ أعمارهم 8 - 29 سنة.",
    "approver": "بيانات ولي الأمر",
    "approverInfoOver18Text": "لا نتطلب من الذين تزيد أعمارهم عن 18 سنة إضافة بيانات الاتصال للوالدين. ولكن إذا كنت تشعر لسبب أو لآخر أنه هناك حاجة للتواصل عند الحاجة مع ولي أمرك، فبإمكانك إضافة بيانات التواصل هنا.",
    "approverInfoText": "يُرسل طلب تأكيد عضويتك إلى هذا العنوان، حيث يتأكد ولي الأمر من البيانات ويُوافق على العضوية.",
    "approveTerms": "لقد اطلعت <0> على السرد بشأن التسجيل لهذه الخدمة </0> <1>وعلى السرد بشأن التسجيل للملف التعريفي لهلسنكي</1> <2>وعلى المعاملات المتعلقة بحماية المعلومات للمدينة</2>. من الممكن التواصل معي من خلال المعلومات التي قدمتها.",
    "basicInfo": "البيانات الأساسية",
    "birthdayHelper": "تاريخ الميلاد (مُفصّلاً)*",
    "birthdayInvalid": "يتوجب أن يكون تاريخ الميلاد بصيغة يوم يوم . شهر شهر . سنة سنة سنة سنة",
    "cancel": "ألغي",
    "childBirthDay": "تاريخ الميلاد",
    "childBirthMonth": "شهر الميلاد",
    "childBirthYear": "سنة الولادة",
    "city": "المدينة",
    "confirmSend": "وافق على الشروط وأرسل الطلب",
    "country": "الدولة",
    "date": "اليوم",
    "dateInvalid": "التاريخ لا يصلح",
    "email": "عنوان البريد الإلكتروني",
    "firstName": "الاسم الشخصي",
    "languageAtHome": "لغة التواصل معك التي في المقام الأول",
    "lastName": "اسم العائلة",
    "membershipInfoText": "العضوية مجانية، وهي لا تُلزمك بأي شيء، ولكن الحصول على البطاقة يعتبر شرطا للمشاركة في فعاليات أماكن الشباب. لا نُسلم بيانات الاتصالات التي قدمتها إلى جهات أخرى لديك الحق أيضاً في التحقق من البيانات التي تمسك وإزالتها. تذكّر أن فعاليات خدمات الشباب خالية من المسكرات والمخدرات.",
    "month": "شهر",
    "pageTitle": "أنشئ ملفا شخصيا",
    "phoneNumber": "رقم الهاتف",
    "photoUsageApproved": "الإذن بالتصوير",
    "photoUsageApprovedNo": "لا",
    "photoUsageApprovedText": "من الممكن التقاط صور لي ويجوز استخدامها أثناء التواصل والتسويق لمدينة هلسنكي في وسائل الإعلام المطبوعة والإلكترونية. لا يُدفع تعويض عن التصوير أو استخدام الصور. الحق في استخدام الصور يكون ساري المفعول دائمًا، ويمس كل التواصل لمدينة هلسنكي والتواصل بخصوص التسويق.",
    "photoUsageApprovedYes": "نعم",
    "postalCode": "الرمز البريدي",
    "processInfoText": "نُرسل الاسئلة بخصوص التأكيد على المعلومات إلى الوالد/الوالدة، وعندما يتحقق من البيانات ويُوافق عليها، فإن عضويتك تُصبح سارية المفعول.",
    "profileLanguage": "لغة المعاملات التي في المقام الأول",
    "remove": "احذف",
    "save": "احفظ المعلومات",
    "schoolClass": "الصف",
    "schoolName": "المدرسة",
    "sendButton": "احفظ المعلومات وأرسل طلب التأكيد",
    "title": "قُم بتعبئة بياناتك",
    "year": "السنة"
  },
  "registry": {
    "descriptionLink": "https://www.hel.fi/static/liitteet-2019/Kaupunginkanslia/Rekisteriselosteet/Kuva/Nuorisopalvelujen%20j%C3%A4senrekisteri.pdf"
  },
  "sanityCheck": "youth-profile.fi",
  "tos": {
    "message1": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    "message2": "",
    "message3": "",
    "message4": "",
    "message5": "",
    "message6": "",
    "message7": "",
    "message8": "",
    "title": "شروط الاستخدام",
    "title1": "1. مجال التطبيق",
    "title2": "2. إنشاء الملف الشخصي وإزالته من الاستخدام",
    "title3": "3. استخدام الملف الشخصي",
    "title4": "4. مدى توفر الخدمة",
    "title5": "5. البيانات الشخصية",
    "title6": "6. الحقوق",
    "title7": "7. تقييد المسؤولية",
    "title8": "8. الشروط الأخرى",
    "updated": "تم تحديثه آخر مرة بتاريخ 20/1/2020"
  },
  "validation": {
    "email": "قم بتعبئة عنوان بريد إلكتروني قيد الاستعمال",
    "invalidPostalCode": "قم بتعبئة الرمز البريدي الصحيح",
    "required": "قم بتعبئة {{label}}",
    "requiredTerms": "وافق على الشروط",
    "tooLong": "قم بتعبئة {{label}}، حيث أنه كحد أقصى {{max}} ضربة زر",
    "tooShort": "قم بتعبئة {{label}}، حيث أنه كحد أدنى {{min}} ضربة زر"
  },
  "youthProfile": {
    "approverEmail": "البريد الإلكتروني",
    "approverInfo": "بيانات ولي الأمر",
    "approverName": "الاسم",
    "approverPhone": "الهاتف",
    "birthdate": "تاريخ الميلاد",
    "homeLanguages": "لغة التواصل معك التي في المقام الأول",
    "school": "المدرسة"
  }
}