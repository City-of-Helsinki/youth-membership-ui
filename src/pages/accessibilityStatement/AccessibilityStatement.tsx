import React from 'react';
import { useTranslation } from 'react-i18next';

import getLanguageCode from '../../common/helpers/getLanguageCode';
import PageWrapper from '../../common/wrapper/PageWrapper';
import styles from './AccessibilityStatement.module.css';
import Header from '../../common/header/Header';
import Footer from '../../common/footer/Footer';
import AccessibilityStatementFi from './AccessibilityStatementFi';
import AccessibilityStatementEn from './AccessibilityStatementEn';
import AccessibilityStatementSv from './AccessibilityStatementSv';

function AccessibilityStatement() {
  const { i18n } = useTranslation();
  const selectStatement = () => {
    const lang = getLanguageCode(i18n.languages[0]);

    switch (lang) {
      case 'fi':
        return <AccessibilityStatementFi />;
      case 'en':
        return <AccessibilityStatementEn />;
      case 'sv':
        return <AccessibilityStatementSv />;
      default:
        return <p>Invalid language.</p>;
    }
  };
  return (
    <PageWrapper>
      <div className={styles.pageWrapper}>
        <Header />
        <div className={styles.container}>
          <div className={styles.innerWrapper}>{selectStatement()}</div>
        </div>
        <Footer />
      </div>
    </PageWrapper>
  );
}

export default AccessibilityStatement;
