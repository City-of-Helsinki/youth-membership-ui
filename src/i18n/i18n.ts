import i18n from 'i18next';
import {initReactI18next} from 'react-i18next';

import en from './en.json';
import fi from './fi.json';
import sv from './sv.json';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: en,
      },
      fi: {
        translation: fi,
      },
      sv: {
        translation: sv,
      },
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false
    }
  });

export default i18n;
