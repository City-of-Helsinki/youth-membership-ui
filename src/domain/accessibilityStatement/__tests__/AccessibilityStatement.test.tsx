import React from 'react';
import { render } from '@testing-library/react';

import AccessibilityStatement from '../AccessibilityStatement';
import i18n from '../../../common/test/testi18nInit';

test('matches snapshot', () => {
  const wrapper = render(<AccessibilityStatement />);
  expect(wrapper.container).toMatchSnapshot();
});

describe('renders correct component based on language', () => {
  test('language is finnish', () => {
    const { getByText } = render(<AccessibilityStatement />);

    expect(getByText('Saavutettavuusseloste')).toBeInTheDocument();
  });

  test('language is english', () => {
    i18n.changeLanguage('en');
    const { getByText } = render(<AccessibilityStatement />);

    expect(getByText('Accessibility Statement')).toBeInTheDocument();
  });

  test('language is swedish', () => {
    i18n.changeLanguage('sv');
    const { getByText } = render(<AccessibilityStatement />);

    expect(getByText('Tillgänglighetsutlåtande')).toBeInTheDocument();
  });

  test('language is something random', () => {
    i18n.changeLanguage('asd');
    const { getByText } = render(<AccessibilityStatement />);

    expect(getByText('Invalid language.')).toBeInTheDocument();
  });
});
