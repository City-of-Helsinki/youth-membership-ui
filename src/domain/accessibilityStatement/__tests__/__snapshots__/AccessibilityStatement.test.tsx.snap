// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <div
    className="background"
  />
  <PageContent>
    <div
      className="innerWrapper"
    >
      <AccessibilityStatementFi />
    </div>
  </PageContent>
</Fragment>
`;
